### PR TITLE
Add azure spring cloud dependencies

### PR DESCRIPTION
**Pull request by moarychan**
Management of all the spring cloud dependencies for azure spring cloud, [here ](https://github.com/microsoft/spring-cloud-azure/blob/1.2.7/spring-cloud-azure-dependencies/pom.xml) is the old version.

Changelog:
- Update `com.azure` group id to `com.azure.spring`.
- Update `spring-cloud-azure-dependencies` artifact id to `azure-spring-cloud-dependencies`.
- Remove `spring-boot-dependencies` and `spring-cloud-dependencies` dependencies.
- Add version tag for versioning upgrade.

### CODE REVIEW

**Comment by moarychan** [comment]
Hi @saragluna @backwind1233 please help to review, thanks.

File: sdk/boms/azure-spring-cloud-dependencies/pom.xml [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        <scope>import</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>org.springframework.cloud</groupId>
+        <artifactId>spring-cloud-dependencies</artifactId>
+        <version>${spring.cloud.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
```
Comment (saragluna) [inline sdk/boms/azure-spring-cloud-dependencies/pom.xml:78]: Should we include this?

Comment (moarychan) [reply sdk/boms/azure-spring-cloud-dependencies/pom.xml:78]: It might be better to separate, let me remove it.

**Comment by backwind1233** [comment]
#18434

**Comment by backwind1233** [comment]
#20386

**Comment by backwind1233** [comment]
feature #20386

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**